libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1086,9 +1091,48 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 		solAssert(false, "Index access only allowed for mappings or arrays.");
 }
 
-void IRGeneratorForStatements::endVisit(IndexRangeAccess const&)
-{
-	solUnimplementedAssert(false, "Index range accesses not yet implemented.");
+void IRGeneratorForStatements::endVisit(IndexRangeAccess const& _indexRangeAccess) {
+	Type const& baseType = *_indexRangeAccess.baseExpression().annotation().type;
+	solUnimplementedAssert(
+		baseType.category() == Type::Category::Array || baseType.category() == Type::Category::ArraySlice,
+		"Index range accesses is available only on arrays and array slices."
+	);
+
+	ArrayType const& arrayType =
+		baseType.category() == Type::Category::Array ?
+		dynamic_cast<ArrayType const &>(baseType) :
+		dynamic_cast<ArraySliceType const &>(baseType).arrayType();
+
+	switch (arrayType.location())
+	{
+		case DataLocation::CallData:
+		{
```

Comment:
Better to assert ``baseType.isDynamicallySized()`` here as well.

Reply:
Fixed